diesel/src/types/mod.rs

```diff
@@ -334,17 +348,38 @@ pub use pg::types::sql_types::*;
 #[cfg(feature = "mysql")]
 pub use mysql::types::*;
 
+/// Indicates that a SQL type exists for a backend.
 pub trait HasSqlType<ST>: TypeMetadata {
+    /// Fetch the metadata for the given type
+    ///
+    /// This method may use `lookup` to do dynamic runtime lookup. Implementors
+    /// of this method should not do dynamic lookup unless absolutely necessary
```

Comment:
Maybe this should include a example case where dynamic lookup is appropriated? (Some sentence mentioning custom sql type definitions?)

Reply:
I think that should be left to the documentation for the backend specific lookup types. I prefer not to put PG specifics here.